lib/timers.js

```diff
@@ -276,7 +276,7 @@ exports.setInterval = function(callback, repeat) {
     timer._repeat();
 
     // Timer might be closed - no point in restarting it
-    if (!timer._repeat)
+    if ((!timer._repeat) || ((timer._idleTimeout === -1) && (!this._handle)))
```

Comment:
There's two extra sets or parenthesis here.

Also, could you make the change to prioritize `timer._idleTimeout === -1`? Thanks!